src/components/products/product-card.svelte

```diff
@@ -21,7 +21,7 @@
       : '/images/create-product/placeholder.svg'
   );
 
-  $: outOfStock = varieties.reduce((sum, variety) => variety.amount + sum, 0) === 0;
+  $: outOfStock = varietiesByColor.get(selectedColor)[0].amount === 0;
```

Comment:
Doesn't this mean "if the first size (can be any) of the selected color is out of stock, display the notice"?
Perhaps we'd be better off keeping the reduce:
```js
varietiesByColor.get(selectedColor).reduce((sum, variety) => variety.amount + sum, 0) === 0
```